src/bun.js/api/bun/socket.zig

```diff
@@ -2044,12 +2045,13 @@ fn NewSocket(comptime ssl: bool) type {
             if (size <= 0) {
                 return JSValue.jsUndefined();
             }
-            var buffer = bun.default_allocator.alloc(u8, @intCast(usize, size)) catch unreachable;
-            defer bun.default_allocator.free(buffer);
-            var tmp = @ptrCast([*c]u8, buffer.ptr);
-            _ = BoringSSL.i2d_SSL_SESSION(session, &tmp);
-            const result = JSC.ArrayBuffer.createBuffer(globalObject, buffer);
-            return result;
+
+            const buffer_size = @intCast(usize, size);
+            var buffer = JSValue.createBufferFromLength(globalObject, buffer_size);
+            var buffer_ptr = @ptrCast([*c]u8, buffer.asArrayBuffer(globalObject).?.ptr);
+
+            _ = BoringSSL.i2d_SSL_SESSION(session, &buffer_ptr);
```

Comment:
should we assert

Reply:
hmm is better to assert this, I will update

Reply:
fix on https://github.com/oven-sh/bun/pull/3596/commits/a0e4a35a1697dadbdf04f0606bd2b37f9b450415